src/get_target_data.py

```diff
@@ -0,0 +1,481 @@
+    param = Option(["--collection-min-date"])
+    result = set_collection_min_date(ctx, param, None)
+    assert result == datetime(2024, 9, 1, 23, 59, 59, tzinfo=timezone.utc)
+
+    # default collection_max_date is 10 days after the nowcast date
+    param = Option(["--collection-max-date"])
+    result = set_collection_max_date(ctx, param, None)
+    assert result == datetime(2024, 10, 12, 23, 59, 59, tzinfo=timezone.utc)
+
+
+def test_bad_inputs():
+    """Bad inputs should return a non-zero exit code."""
+    runner = CliRunner()
+
+    # sequence_as_of cannot be in the future
+    with runner.isolated_filesystem():
+        result = runner.invoke(
+            main,
+            [
+                "--nowcast-date",
+                "2024-10-02",
+                "--sequence-as-of",
+                "2054-10-02",
+            ],
+            catch_exceptions=True,
+            color=True,
+            standalone_mode=False,
+        )
+        assert result.exit_code == 1
+
+    # nowcast_date must match a variant-nowcast-hub round_id
+    with runner.isolated_filesystem():
+        result = runner.invoke(
+            main,
+            [
+                "--nowcast-date",
+                "2022-2-2",
+                "--sequence-as-of",
+                "2024-10-01",
+            ],
+            catch_exceptions=True,
+            color=True,
+            standalone_mode=False,
+        )
+        assert result.exit_code == 1
+
+
+def test_target_data(caplog, tmp_path):
+    """
+    If the modeled-clades file doesn't have meta.created_at, tree_as_of should default to
+    nowcast_date - two days.
+    """
+    caplog.set_level(logging.INFO)
+
+    nowcast_date = "2024-09-11"
+    runner = CliRunner()
+    result = runner.invoke(
+        main,
+        [
+            "--nowcast-date",
+            nowcast_date,
+            "--target-data-dir",
+            tmp_path,
+        ],
+        color=True,
+        catch_exceptions=False,
+        standalone_mode=False,
+    )
+    assert result.exit_code == 0
+    ts = pl.read_parquet(result.return_value[0])
+
+    # sequence date should default to nowcast_date + 90 days
+    assert "sequence_as_of=2024-12-10" in caplog.text
+    # tasks.json for 2024-09-11 doesn't have a meta.created_at field, so tree_as_of = nowcast_date - 2 days
+    assert "tree_as_of=2024-09-09" in caplog.text
+
+    # time series target dates should be limited to dates that match collection date min/max options
+    # (2024-08-11 to 2024-09-21 is 43 days, but both have 11:59:59 timestamps, so we'd expect
+    # 42 days in the time series)
+    target_dates = ts["target_date"].unique().to_list()
+    assert len(target_dates) == 42
+
+    modeled_clades_path = Path("auxiliary-data/modeled-clades") / f"{nowcast_date}.json"
+    modeled_clades = json.loads(modeled_clades_path.read_text(encoding="utf-8"))
+    round_clades = [
+        clade for clade in modeled_clades.get("clades", []) if clade != "other"
+    ]
```

Comment:
And we'll want to keep "other" here as well:
```suggestion
    round_clades = modeled_clades.get("clades", [])
```